gdnative-core/src/variant.rs

```diff
@@ -490,32 +519,20 @@ impl Variant {
         unsafe { (get_api().godot_variant_has_method)(&self.0, &method.0) }
     }
 
-    // TODO: return a proper error.
-    pub fn call(&mut self, method: &GodotString, args: &[Variant]) -> Result<(), ()> {
+    pub fn call(&mut self, method: &GodotString, args: &[Variant]) -> Result<Variant, CallError> {
         unsafe {
             let api = get_api();
             let mut err = sys::godot_variant_call_error::default();
-            if args.is_empty() {
-                let mut first = ::std::ptr::null() as *const sys::godot_variant;
-                (api.godot_variant_call)(&mut self.0, &method.0, &mut first, 0, &mut err);
-            } else {
-                // TODO: double check that this is safe.
-                let gd_args: &[sys::godot_variant] = transmute(args);
-                let mut first = &gd_args[0] as *const sys::godot_variant;
-                (api.godot_variant_call)(
-                    &mut self.0,
-                    &method.0,
-                    &mut first,
-                    args.len() as i32,
-                    &mut err,
-                );
-            }
-
-            if err.error == sys::godot_variant_call_error_error_GODOT_CALL_ERROR_CALL_OK {
-                Ok(())
-            } else {
-                Err(())
-            }
+            let mut arg_refs = args.iter().map(Variant::sys).collect::<Vec<_>>();
```

Comment:
Hmm, this will always, I guess there is no sane way around that?

Reply:
We can take a `&mut [&Variant]` instead, then I think we can get a `*mut *const sys::godot_variant` out of it without an allocation *here*. I don't think people usually keep `&mut [&Variant]`s around, though. In most cases the caller will likely have to allocate anyway, and the interface would be more complicated.